src/main/groovy/org/pih/warehouse/product/ProductSupplierDetailsCommand.groovy

```diff
@@ -0,0 +1,43 @@
+package org.pih.warehouse.product
+
+import grails.validation.Validateable
+import org.pih.warehouse.core.Organization
+import org.pih.warehouse.core.RatingTypeCode
+
+class ProductSupplierDetailsCommand implements Validateable {
+
+    Product product
+
+    Organization supplier
+
+    Organization manufacturer
+
+    String code
+
+    String name
+
+    String supplierCode
+
+    String manufacturerCode
+
+    Boolean active
+
+    String description
+
+    String brandName
+
+    RatingTypeCode ratingTypeCode
+
+    String productCode
+
+    static constraints = {
+        code(nullable: true)
+        manufacturer(nullable: true)
+        ratingTypeCode(nullable: true)
+        description(nullable: true)
+        manufacturerCode(nullable: true)
+        brandName(nullable: true)
+        productCode(nullable: true)
+        active(nullable: true)
+    }
+}
```

Comment:
Since we are about to create 3 separate endpoints for now, I've decided to go with the command approach to make sure, that when using details endpoint, a user will potentially be able only to edit details part and that the user won't be able to include any other properties, e.g. throwing a list of `productSupplierPreferences` that would be persisted if I used the `ProductSupplier` domain as a command class.

cc @awalkowiak @alannadolny @drodzewicz @jmiranda

Reply:
I wonder if we could somehow use shared constraints, like here: https://grails.github.io/legacy-grails-doc/3.3.0/ref/Constraints/Usage.html so we have a "single source of truth" regarding constraints, and won't have to change two places whenever constraints will change on the domain level.

Reply:
@awalkowiak thanks, this feels interesting. I have not known this was possible. I will check it out and get back to you.

Reply:
I found an even worse issue than what @kchelstowski was worried about. And we are definitely susceptible to this given that none of our domain or command constraints are marked as final.

https://grails.org/blog/2022-09-28-static-binding-bug.html

Reply:
@awalkowiak Regarding your comment about shared constaints, which class should be the single source of truth.

Reply:
I'm guessing the domain in this case, and then we could override it if needed.